addons/point_of_sale/models/pos_session.py

```diff
@@ -313,6 +302,522 @@ def action_pos_session_close(self):
+                'currency_id': session_currency.id,
+                **init_vals
+            }
+
+    @api.model
+    def _debit_amounts(self, partial_move_line_vals, amount, amount_converted, session_currency, is_using_company_currency):
+        """ `partial_move_line_vals` is completed by `debit`ing the given amounts.
+
+        See _credit_amounts docs for more details.
+        """
+        if is_using_company_currency:
+            if amount > 0:
+                return {'credit': 0.0, 'debit': amount, **partial_move_line_vals}
+            else:
+                return {'credit': abs(amount), 'debit': 0.0, **partial_move_line_vals}
+        else:
+            if amount_converted > 0:
+                init_vals = {'credit': 0.0, 'debit': amount_converted, **partial_move_line_vals}
+            else:
+                init_vals = {'credit': abs(amount_converted), 'debit': 0.0, **partial_move_line_vals}
+            return {
+                'amount_currency': amount if amount_converted > 0 else -amount,
+                'currency_id': session_currency.id,
+                **init_vals
+            }
+
+    @api.multi
+    def show_journal_items(self):
+        def get_matched_move_lines(aml):
+            if aml.credit > 0:
+                return [r.debit_move_id.id for r in aml.matched_debit_ids]
+            else:
+                return [r.credit_move_id.id for r in aml.matched_credit_ids]
+
+        # get all the linked move lines to this account move.
+        move = self.move_id
+        non_reconcilable_lines = move.line_ids.filtered(lambda aml: not aml.account_id.reconcile)
+        reconcilable_lines = move.line_ids - non_reconcilable_lines
+        fully_reconciled_lines = reconcilable_lines.filtered(lambda aml: aml.full_reconcile_id)
+        partially_reconciled_lines = reconcilable_lines - fully_reconciled_lines
+
+        other_cash_move_lines = self.env['account.move.line'].search_read([('statement_id', '!=', False), ('statement_id', '=', self.cash_register_id.id)], ['id'])
+        other_cash_move_lines_ids = [line['id'] for line in other_cash_move_lines]
+
+        ids = (non_reconcilable_lines.ids
+                + fully_reconciled_lines.mapped('full_reconcile_id').mapped('reconciled_line_ids').ids
+                + sum(partially_reconciled_lines.mapped(get_matched_move_lines), partially_reconciled_lines.ids)
+                + other_cash_move_lines_ids)
+
+        all_related_moves = self.env['account.move.line'].browse(ids).mapped('move_id')
+
+        # call the account move line action tree view with default filter (account)
+        # and domain containing the calculated ids above
+        [action] = self.env.ref('account.action_account_moves_all_a').read()
+        context = safe_eval(action['context'])
+        context.pop('search_default_sales')  # remove Sales default filter
+        action['context'] = context
+        action['domain'] = [('id', 'in', all_related_moves.mapped('line_ids').ids)]
```

Comment:
return {
'name': 'Journal Items',
context: {'journal_type':'general', ....},
'res_model': 'account.move.line',
'domain': [('id', 'in', all_related_moves.mapped('line_ids').ids)],
'view_id':  'view_move_line_tree_grouped',
'view_mode': 'tree,pivot,graph,form,kanban',
}